tests/test_nats_sts.py

```diff
@@ -0,0 +1,119 @@
+from tests.helm_template_generator import render_chart
+import pytest
+from . import supported_k8s_versions
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestNatsStatefulSet:
+    def test_nats_statefulset_defaults(self, kube_version):
+        """Test that nats statefulset is good with defaults."""
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only=["charts/nats/templates/statefulset.yaml"],
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        c_by_name = {
+            c["name"]: c for c in doc["spec"]["template"]["spec"]["containers"]
+        }
+        assert doc["kind"] == "StatefulSet"
+        assert doc["apiVersion"] == "apps/v1"
+        assert doc["metadata"]["name"] == "RELEASE-NAME-nats"
+        assert c_by_name["metrics"]["image"].startswith(
+            "quay.io/astronomer/ap-nats-exporter:"
+        )
+        assert c_by_name["nats"]["image"].startswith(
+            "quay.io/astronomer/ap-nats-server:"
+        )
+        assert c_by_name["nats"]["livenessProbe"] == {
+            "httpGet": {"path": "/", "port": 8222},
+            "initialDelaySeconds": 10,
+            "timeoutSeconds": 5,
+        }
+        assert c_by_name["nats"]["readinessProbe"] == {
+            "httpGet": {"path": "/", "port": 8222},
+            "initialDelaySeconds": 10,
+            "timeoutSeconds": 5,
+        }
+
+        assert doc["spec"]["template"]["spec"]["nodeSelector"] == {}
+        assert doc["spec"]["template"]["spec"]["affinity"] == {}
+        assert doc["spec"]["template"]["spec"]["tolerations"] == []
+
+    def test_nats_statefulset_with_metrics_and_resources(self, kube_version):
+        """Test that nats statefulset renders good metrics exporter."""
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only=["charts/nats/templates/statefulset.yaml"],
+            values={
+                "nats": {
+                    "exporter": {
+                        "enabled": True,
+                        "resources": {"requests": {"cpu": "234m"}},
+                    },
+                    "nats": {"resources": {"requests": {"cpu": "123m"}}},
+                },
+            },
+        )
+
+        assert len(docs) == 1
+        containers = docs[0]["spec"]["template"]["spec"]["containers"]
+        assert len(containers) == 2
+        c_by_name = {c["name"]: c for c in containers}
+        assert c_by_name["nats"]["resources"]["requests"]["cpu"] == "123m"
+        assert c_by_name["metrics"]["resources"]["requests"]["cpu"] == "234m"
+
+    def test_nats_statefulset_with_affinity_and_tolerations(self, kube_version):
+        """Test that nats statefulset renders proper nodeSelector, affinity, and tolerations"""
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only=["charts/nats/templates/statefulset.yaml"],
+            values={
```

Comment:
Maybe you could also add a test with the .Values.global default values for nodeSelector, affinity and tolerations ?

As you've added the behaviour to default to these values if chart-level values are not provided.